comdb2/dbapi2.py

```diff
@@ -114,17 +119,59 @@
 In this example, we run the query with two different sets of
 parameters, producing different results.  First, we execute the query with
 parameter ``a`` bound to ``20`` and ``b`` bound to ``42``.  In this case,
-because ``20 <= 25 <= 42``, the expression evaluates to true, and a ``1`` is
-returned.
+because ``20 <= 25 <= 42``, the expression evaluates to true, and a row
+containing a single column with a value of ``1`` is returned.
 
-When we run the same query with parameter ``b`` bound to ``23``, a ``0`` is
-returned instead, because ``20 <= 25 <= 23`` is false.
+When we run the same query with parameter ``b`` bound to ``23``, a row
+containing a single column with a value of ``0`` is returned instead, because
+``20 <= 25 <= 23`` is false.
 
-Note:
-    Because parameters are bound using ``%(name)s``, other ``%`` signs in
+Warning:
+    Because named parameters are bound using ``%(name)s``, other ``%`` signs in
     a query must be escaped.  For example, ``WHERE name like 'M%'`` becomes
     ``WHERE name LIKE 'M%%'``.
 
+Danger:
+    This applies even when no parameters are passed at all - passing ``None``
+    or no parameters behaves the same as passing an empty `dict`.
+
+By Position
+~~~~~~~~~~~
+
+You can bind parameters positionally rather than by name, by using ``?``
+for each placeholder and providing a list or tuple of parameter values.
+For example:
+
+    >>> query = "select 25 between ? and ?"
+    >>> print(conn.cursor().execute(query, [20, 42]).fetchall())
+    [[1]]
+
+In this example, we execute the query with the first ``?`` bound to 20 and the
+second ``?`` bound to 42, so a row with a single column with a value of ``1``
+is returned like in the previous example.
+
+Warning:
+    Unlike when binding parameters by name, you must not escape ``%`` signs in
+    the SQL when binding parameters positionally.
+
+    For example, you could do ``WHERE val % 5 = ?``, but not ``WHERE val %% 5 = ?``.
+
+Tip:
+    You can pass an empty tuple of parameters to avoid the need to escape ``%``
+    signs in the SQL even when you don't want to bind any parameters, like:
+
+    >>> query = "select 42 % 20"
+    >>> print(conn.cursor().execute(query, ()).fetchall())
```

Comment:
Wait, does that work? I thought `()` wouldn't create an empty tuple, and that `(,)` or `tuple()` was needed instead.

Reply:
Not only is it not required to do `(,)`, but in fact `(,)` is a syntax error!

For non-empty tuples, it's a comma that makes the tuple, and the parens are optional, but for _empty_ tuples, the literal is a pair of parentheses with nothing between them.